docs/reference/data-streams/set-up-tsds.asciidoc

```diff
@@ -0,0 +1,349 @@
+          }
+        }
+      },
+      "cold": {
+        "min_age": "60d",
+        "actions": {
+          "searchable_snapshot": {
+            "snapshot_repository": "found-snapshots"
+          }
+        }
+      },
+      "frozen": {
+        "min_age": "90d",
+        "actions": {
+          "searchable_snapshot": {
+            "snapshot_repository": "found-snapshots"
+          }
+        }
+      },
+      "delete": {
+        "min_age": "735d",
+        "actions": {
+          "delete": {}
+        }
+      }
+    }
+  }
+}
+----
+
+[discrete]
+[[tsds-create-mappings-component-template]]
+==== Create a mappings component template
+
+A TSDS requires a matching index template. In most cases, you compose this index
+template using one or more component templates. You typically use separate
+component templates for mappings and index settings. This lets you reuse the
+component templates in multiple index templates.
+
+For a TSDS, the mappings component template must include mappings for:
+
+* One or more <<time-series-dimension,dimension fields>> with a
+`time_series_dimension` value of `true`. At least one of these dimensions must
+be a plain `keyword` field.
+
+Optionally, the template can also include mappings for:
+
+* One or more <<time-series-metric,metric fields>>, marked using the
+`time_series_metric` mapping parameter.
+
+* A `date` or `date_nanos` mapping for the `@timestamp` field. If you don’t
+specify a mapping, Elasticsearch maps `@timestamp` as a `date` field with
+default options.
+
+[source,console]
+----
+PUT _component_template/my-weather-sensor-mappings
+{
+  "template": {
+    "mappings": {
+      "properties": {
+        "sensor_id": {
+          "type": "keyword",
+          "time_series_dimension": true
+        },
+        "location": {
+          "type": "keyword",
+          "time_series_dimension": true
+        },
+        "temperature": {
+          "type": "half_float",
+          "time_series_metric": "gauge"
+        },
+        "humidity": {
+          "type": "half_float",
+          "time_series_metric": "gauge"
+        },
+        "@timestamp": {
+          "type": "date",
+          "format": "strict_date_optional_time"
+        }
+      }
+    }
+  },
+  "_meta": {
+    "description": "Mappings for weather sensor data"
+  }
+}
+----
+// TEST[continued]
+
+[discrete]
+[[tsds-create-index-settings-component-template]]
+==== Create an index settings component template
+
+Optionally, the index settings component template for a TSDS can include:
+
+* Your lifecycle policy in the `index.lifecycle.name` index setting.
+* The <<tsds-look-ahead-time,`index.look_ahead_time`>> index setting.
+* Other index settings, such as <<index-codec,`index.codec`>>, for your TSDS's
+backing indices.
+
+IMPORTANT: Don't specify the `index.routing_path` index setting in a component
+template. You must configure `index.routing_path` directly in the index
+template. You'll do this in the following step.
```

Comment:
`routing_path` will now default to all of the `keyword` `time_series_dimension` fields in the template. This is *probably* what folks want. At least if they declare the entire mapping in their template.

Reply:
👍 
I updated the `index.routing_path` definition on the [Time series index settings page](https://tsds001.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/tsds-index-settings.html) page to include `Defaults to the list of <<time-series-dimension,dimension fields>> with a
`time_series_dimension` value of `true` defined in your component templates.`